src/main/java/com/dataloom/neuron/audit/AuditLogQueryService.java

```diff
@@ -24,23 +44,25 @@
     private final Session           session;
     private final PreparedStatement storeQuery;
 
-    public AuditLogQueryService( String keyspace, Session session ) {
+    // TODO: what would CassandraConfiguration be needed for in the future?
+    public AuditLogQueryService( CassandraConfiguration cassandraConfig, Session session ) {
 
         this.session = session;
         this.storeQuery = session.prepare( Table.AUDIT_LOG.getBuilder().buildStoreQuery() );
     }
 
     public void store( AuditableSignal signal ) {
 
+        // TODO: how do I deal with Optionals?
```

Comment:
second Katherine's question, and similarly when is principal absent? 
1. `Optional.get()` would throw `IllegalStateException` when the value is absent. So you are really assuming that it must be present already. But in that case, why do we need `Optional`?
2. Also, Cassandra wouldn't allow partition key/clustering key to be null.